packages/core/saved-objects/core-saved-objects-migration-server-internal/src/core/build_active_mappings.ts

```diff
@@ -25,18 +24,17 @@ import type {
  * @param typeDefinitions - the type definitions to build mapping from.
  */
 export function buildActiveMappings(
-  typeDefinitions: SavedObjectsTypeMappingDefinitions | SavedObjectsMappingProperties
+  typeDefinitions: SavedObjectsTypeMappingDefinitions | SavedObjectsMappingProperties,
+  override?: Partial<IndexMapping>
 ): IndexMapping {
   const mapping = getBaseMappings();
 
   const mergedProperties = validateAndMerge(mapping.properties, typeDefinitions);
 
   return cloneDeep({
     ...mapping,
+    ...override,
     properties: mergedProperties,
-    _meta: {
-      migrationMappingPropertyHashes: md5Values(mergedProperties),
-    },
```

Comment:
At _KibanaMigrator_ level, we're building the active mappings to expose that property to other consumers (e.g. _SavedObjects_ repository). These consumers don't currently need the `_meta` information, so there's times where we're only interested in the _mappings.properties_.

On the other hand, when building actual mappings for each index, there's some of the `_meta` info which is common to all indices, and some other that is index-specific.

Before the PR, the `md5values(...)` were calculated once for all SO types, and then calculated again for each index's types. That is/was a duplicated effort for an expensive operation. I intend to avoid this, by freeing `buildActiveMappings` of this responsibility.

I agree that perhaps I can separate this in 2 different methods (one to merge root fields + types, and another one to clone + add meta), and that the one that receives Partial<IndexMapping> could receive `IndexMappingMeta`.